crates/app/src/lib.rs

```diff
@@ -207,6 +207,22 @@ impl<'a, L> App<'a, L> {
             .map(|locked| AppTrigger { app: self, locked })
     }
 
+    /// Returns the trigger metadata for a specific trigger type.
+    pub fn get_trigger_metadata<'this, T: Deserialize<'this> + Default>(
+        &'this self,
+        trigger_type: &'a str,
+    ) -> Result<T> {
+        Some(&self.locked.metadata["triggers"][trigger_type])
```

Comment:
This will panic if `"triggers"` or the specific `trigger_type` don't exist. This whole function should probably return `Result<Option<T>>` instead, with `Ok(None)` if the key(s) are missing.